chain/near.ts

```diff
@@ -0,0 +1,416 @@
+      public outputDataReceivers: Array<DataReceiver>,
+      public inputDataIds: Array<CryptoHash>,
+      public actions: Array<ActionValue>,
+    ) {}
+  }
+
+  export enum SuccessStatusKind {
+    VALUE = 0,
+    RECEIPT_ID = 1,
+  }
+
+  // Doesn't have Value suffix because it has
+  // VALUE variant/kind, that would be confusing.
+  export class SuccessStatus {
+    constructor(
+      public kind: SuccessStatusKind,
+      public data: Payload,
+    ) {}
+
+    toValue(): Bytes {
+      assert(this.kind == SuccessStatusKind.VALUE, 'SuccessStatus is not a \'Value\'.')
+      return changetype<Bytes>(this.data as u32)
+    }
+
+    toReceiptId(): CryptoHash {
+      assert(this.kind == SuccessStatusKind.RECEIPT_ID, 'SuccessStatus is not a \'ReceiptId\'.')
+      return changetype<CryptoHash>(this.data as u32)
+    }
+
+    static fromValue(input: Bytes): SuccessStatus {
+      return new SuccessStatus(SuccessStatusKind.VALUE, changetype<u32>(input))
+    }
+
+    static fromReceiptId(input: CryptoHash): SuccessStatus {
+      return new SuccessStatus(SuccessStatusKind.RECEIPT_ID, changetype<u32>(input))
+    }
+  }
+
+  export enum Direction {
+    LEFT = 0,
+    RIGHT = 1,
+  }
+
+  export class MerklePathItem {
+    constructor(
+      public hash: CryptoHash,
+      public direction: Direction,
+    ) {}
+
+    @operator('<')
+    lt(other: MerklePathItem): boolean {
+      abort("Less than operator isn't supported in MerklePathItem")
+      return false
+    }
+
+    @operator('>')
+    gt(other: MerklePathItem): boolean {
+      abort("Greater than operator isn't supported in MerklePathItem")
+      return false
+    }
+
+    toString(): string {
+      return `{hash: ${this.hash.toString()}}, direction: ${this.direction.toString()}`
+    }
+  }
+
+  export class MerklePath extends Array<MerklePathItem> {}
+
+  export class ExecutionOutcome {
+    constructor(
+      public proof: MerklePath,
+      public blockHash: CryptoHash,
+      public id: CryptoHash,
+      public logs: Array<string>,
+      public receiptIds: Array<CryptoHash>,
+      public gasBurnt: u64,
+      public tokensBurnt: BigInt,
+      public executorId: string,
+      public status: SuccessStatus,
+    ) {}
+  }
+
+  export class SlashedValidator {
+    constructor(
+      public accountId: AccountId,
+      public isDoubleSign: bool,
+    ) {}
+  }
+
+  export class BlockHeader {
+    constructor(
+      public height: BlockHeight,
+      public prevHeight: BlockHeight,// Always zero when version < V3
+      public epochId: CryptoHash,
+      public nextEpochId: CryptoHash,
+      public hash: CryptoHash,
+      public prevHash: CryptoHash,
+      public prevStateRoot: CryptoHash,
+      public chunkReceiptsRoot: CryptoHash,
+      public chunkHeadersRoot: CryptoHash,
+      public chunkTxRoot: CryptoHash,
+      public outcomeRoot: CryptoHash,
+      public chunksIncluded: u64,
+      public challengesRoot: CryptoHash,
+      public timestampNanosec: u64,
```

Comment:
I think the `timestamp` is missing, this is, AFAIK, only the nano sec of the timestamp so it kind of add up to the `timestamp` value in milliseconds (also to be validated) and adds extra precision.

We have the `timestamp` in the protobuf.